src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -172,7 +229,9 @@ public virtual InputFormatterExceptionModelStatePolicy ExceptionPolicy
 
             var request = context.HttpContext.Request;
 
-            if (!request.Body.CanSeek && !_suppressInputFormatterBuffering)
+            var suppressInputFormatterBuffering = _suppressInputFormatterBuffering ?? _options.SuppressInputFormatterBuffering;
```

Comment:
The normal case should be to use the `MvcOptions` constructor. So, reverse the `_suppressInputFormatterBuffering` and `_suppressJsonDeserializationExceptionMessages` checks.
``` c#
var suppressInputFormatterBuffering = _options?.SuppressInputFormatterBuffering ??
 _suppressInputFormatterBuffering.Value;
```